automation/src/main/java/org/wso2/testgrid/automation/parser/TestNgResultsParser.java

```diff
@@ -205,7 +206,16 @@ private String getClassName(StartElement classElement) {
             }
             if (event.getEventType() == XMLStreamConstants.START_ELEMENT && FAILED
                     .equals(event.asStartElement().getName().getLocalPart())) {
-                String failureMessage = readFailureMessage(eventReader);
+                String failureMessage = "unknown";
+                Iterator attributes = event.asStartElement().getAttributes();
+                failureMessage = readFailureMessage(eventReader);
+                while (attributes.hasNext()) {
+                    Attribute attribute = (Attribute) attributes.next();
+                    if (attribute.getName().getLocalPart().equals("message")) {
```

Comment:
Better to change it to ("message").equals(attribute.getName().getLocalPart()) to avoid possible NPEs.